src/polymer/elements/iron-form-v1-to-v2.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import {Document, ParsedHtmlDocument, Replacement, Severity, Warning} from 'polymer-analyzer';
+
+import {HtmlRule} from '../../html/rule';
+import {getIndentationInside} from '../../html/util';
+import {registry} from '../../registry';
+import {stripIndentation} from '../../util';
+
+const p = dom5.predicates;
+const isIronFormV1 = p.AND(
+    (node: dom5.Node) =>
+        !node.parentNode || !p.hasTagName('iron-form')(node.parentNode),
+    p.hasTagName('form'),
+    p.hasAttrValue('is', 'iron-form'));
+
+const propertiesToDelete = [
+  'is',
+  'disable-native-validation-ui',
+  'disable-native-validation-ui$',
+  'request',
+  'request$',
+];
+const propertiesToMove = [
+  'id',
+  'id$',
+  'headers',
+  'headers$',
+  'with-credentials',
+  'with-credentials$'
+];
+const propertiesToRename: {[key: string]: string;} = {
+  'content-type': 'enctype',
+  'content-type$': 'enctype$',
+};
+
+class IronFormV1ToV2 extends HtmlRule {
+  code = 'iron-form-v1-to-v2';
+  description = stripIndentation(`
+      Warns when \`iron-form\` is used as type extension. 
+
+      This:
+
+          <form is="iron-form"
+                method="get"
+                action="/my-end-point"
+                on-iron-form-error="handleError">
+            <input type="text">
+            <input type="submit">
+          </form>
+
+      Should instead be written as:
+
+          <iron-form on-iron-form-error="handleError">
+            <form method="get" action="/my-end-point">
+              <input type="text">
+              <input type="submit">
+            </form>
+          </iron-form>
+  `);
+
+  async checkDocument(parsedDocument: ParsedHtmlDocument, _document: Document) {
+    const warnings: Warning[] = [];
+    const forms = dom5.queryAll(
+        parsedDocument.ast, isIronFormV1, [], dom5.childNodesIncludeTemplate);
+    for (const form of forms) {
+      // Collect attributes/events for the iron-form and form.
+      let ironFormAttrs = '', formAttrs = '';
+      form.attrs.forEach((attr) => {
+        // All `iron-form-*` events and `propertiesToMove` go on <iron-form>.
+        if ((attr.name.indexOf('on-iron-form-') === 0 ||
+             propertiesToMove.indexOf(attr.name) >= 0)) {
+          ironFormAttrs +=
+              ` ${attr.name}${attr.value ? `="${attr.value}"` : ''}`;
```

Comment:
This is battle tested, so it's fine, but for future passes I think we should try to use dom5 APIs as much as possible rather than building HTML with string manipulation. We need more convenience methods for doing this.

It's not a security thing, as this is transforming from trusted code to trusted code, it's just less error prone.